### ISSUE TITLE
State pattern tutorial

### ISSUE THREAD

**Issue by FrankJRix** [godotengine/godot-docs#3313]
Hi, just wanted to point out that the state pattern tutorial here:

https://docs.godotengine.org/en/stable/tutorials/misc/state_design_pattern.html

is completely non-working. I don't have the skill to fix that, beside the most obvious flaws, so I just thought I'd point that out hoping someone could.

**Comment by Calinou** [comment]
Can you point to parts of the code that don't work? There's not much we can do without context :slightly_smiling_face:

**Comment by FrankJRix** [comment]
At first I manually copied the code to reproduce the tutorial and experiment with it, and while typing I noticed a few issues like inputting a vector2 (actually null, since it would be initialized later) to the abs function, and generally not checking if variables are initialized, which didn't allow the code to "complile" at all.
I jury-rigged those and the code still doesn't work as intended, which makes me wonder if the author actually ran it. 

I hope this doesn't come out as whining or anything, I just wanted to point out a flaw in the otherwise perfect documentation.

Thanks for the reply!
